### DISCUSSION TITLE
[Uczymy się architektury] 🚀🚀 Architektura w Azure - Governance #azure-solutions-architect

### DISCUSSION THREAD

**Discussion by maciejgos** [AzureCommunityPL/group#209] [Wydarzenia]
Cześć! 🙂
Nowy rok i lecimy z kolejnym spotkaniem 😀.Na następnym spotkaniu spróbujemy porozmawiać o **Azure Governance**!
Skupimy się na takich technologiach jak:
- Azure Policy,
- Azure Blueprints,
- Azure Landing Zones

Następne spotkanie 25/02/2021 godzina 20:00

🚀🚀🚀 Zadanie 🚀🚀🚀
Firma ACME Inc 😉 chce przenieść swoje zasoby utrzymywane we własnej serwerowni do chmury. Będzie to pierwsza mała testowa migracja nie kluczowych systemów.

Systemy które będą migrowane to:

- 2 aplikacje napisane w NodeJS do HR (rekrutacja) oraz obsługi urlopów
- 1 aplikacja napisana w .NET Framework do obsługi sklepu internetowego
- 1 aplikacja napisana w Delphi - obecnie działa ona na maszynie wirtualnej, jest to aplikacja do obsługi zamówień w firmie (długopisy, papier do drukarek itp.)

Jedynie sklep internetowy powinien być dostępny przez cały miesiąc z możliwym 24h oknem serwisowym w miesiącu.

Powinny być utworzone oddzielne środowiska dla produkcji oraz development, dział finansowy chciałby mieć możliwość śledzenia kosztów poszczególnych usług, oraz projektów jakie będą się toczyły

👉 Przydatne linki
https://docs.microsoft.com/en-us/azure/governance/policy/overview
https://docs.microsoft.com/en-us/azure/governance/blueprints/overview
https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/landing-zone/
https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/landing-zone/migrate-landing-zone

💻 Link do spotkania: https://bit.ly/2YItQOs
📅 Kalendarz: https://calndr.link/event/Xkf0G7jrEW
📖 Poprzednie spotkania (wraz z nagraniami): https://azurecommunity.pl/

Masz pytanie? Napisz do mnie lub @arryl !

O cyklu spotkań #azure-solution-architect:
Uczymy się tworzyć architekturę w Azure! Jeśli myślisz, że to też Twoja ścieżka - dołącz do nas! Zapraszamy osoby o każdym poziomie zaawansowania!
Każde spotkanie to nowe architektoniczne zadanie, które pozwala nam poznać dany fragment Azure - omawiamy nasze propozycje i w ten sposób zdobywamy wiedzę! Bazujemy na wiedzy wymaganej na egzaminach AZ-303 oraz AZ-304, ale głównie skupiamy się na bardziej praktycznych i życiowych problemach.

**Comment by kaluzaaa** [comment]
Ważne - https://docs.microsoft.com/en-us/answers/questions/177308/index.html ;-)

**Reply by maciejgos** [reply]
Dzięki @kaluzaaa :)

**Reply by phaate** [reply]
Jak to w ogóle jest z tymi blueprintami, warto w ogóle je brać pod uwagę na produkcje?

**Reply by aryll** [reply]
a czemu nie? to chyba zależy co robisz i jak robisz 😊

zakładając że masz środowisko developerskie, stagingowe i produkcyjne to pewnie chcesz żeby to co zostało przetestowane na stagingu zachowywało się tak samo na produkcji - i tutaj mogą się przydać Blueprinty moim zdaniem.

Ale oczywiście mogę się mylić :)

**Reply by maciejgos** [reply]
Generalnie Blueprints na stan mojej wiedzy są cały czas w preview ale są traktowane jako GA 😉 oraz tak jak zmieścił link @kaluzaaa obecnie całe siły zostały rzucone na Template Spec. 

> Lastly, even though the Blueprint service is listed as “Preview”, we treat the service as GA, which means even once a migration path is made available, we will continue to support the current Blueprint APIs for 1-3 years or until everyone has been migrated.

Podsumowując moją odpowiedź można by tego użyć ale trzeba mieć z tyłu głowy potencjalną migracje w bliżej nieokreślonej przyszłości. Ostatni punkt już nie koniecznie może być Twoim zmartwieniem 😉

**Reply by aryll** [reply]
Ja mam wrażenie że pomimo preview Blueprints traktowane są jako 'normalna' usługa - i chyba tak szybko nie znikną skoro są nawet częścią AZ 304

**Reply by kaluzaaa** [reply]
@aryll egazminy nie mają nic wspólnego z road mapą usług ;-) 

```
Lastly, even though the Blueprint service is listed as “Preview”, we treat the service as GA, which means even once a migration path is made available, we will continue to support the current Blueprint APIs for 1-3 years or until everyone has been migrated.
```
**migration path is made available, we will continue to support the current Blueprint APIs for 1-3 years or until everyone has been migrated.** ;-)

**Reply by aryll** [reply]
ok, ok
@kaluzaaa ja po prostu nie wierzę w roadmapy

**Reply by KamilWiecek** [reply]
> Jak to w ogóle jest z tymi blueprintami, warto w ogóle je brać pod uwagę na produkcje?

Mają swoje plusy: 
- locki, które może ściągnąć tylko Manged Identity, ale jeśli nie rozdajesz Ownera na lewo i prawo to nic nie zzyskujesz,
- możliwość wersjonowanie w portalu i weryfikacji, która wersja blueprinta jest aktualnie przypisana do subskrypcji,
- kilka gotowców, z których można skorzystać po prostu je parametryzując, albo wyeksportować i dostosować

Też jestem ciekawy tego, czy ktoś używa, wdraża i wersjonuje automatycznie BP na większą skalę. Ktokolwiek?

Sam nie używam, mam za sobą jedną nieudaną próbę wejścia w blueprinty. Rozwiązanie było wtedy niedojrzałe - kojarzę spore braki w dokumentacji. Był rekomendowany powershell module do deploymentu blueprintow as a code, w którym trzeba było pomiędzy plikami hardkodować stringi bo były referencjami. 

Olałem, bo skoro:
- Azure Policy(Set) Assignments można zapiąć na poziomie MG i masz załawtioną sprawę + nie masz definicji w wielu miejscach,
- i tak trzeba używać ARM Templatów, 
to dlaczego po prostu nie pozostać przy nich? Ogrom przykładów i można wdrażać cross-scope.

Wadą jest, zarządzanie cyklem życia subskrypcji po początkowym wdrożeniu. Ale... to już zależy, czy wolisz zbudować proces i wdrażąc nowe wersja ARMów (tu Template Spec Ci się przyda), czy ogarniasz temat korzystając z Az Policy. 

EDIT:
uzupełniłem trochę informacje dzięki linkowi od @kaluzaaa, powiem tak... jakby mi czytali w myślach :)

**Comment by krukowskid** [comment]
Gdzie można zobaczyć nagranie ze spotkania?

**Reply by aryll** [reply]
@krukowskid będzie wkrótce - dam znać jak tylko będzie dostępne 😊

**Reply by aryll** [reply]
Jest już dostępna notatka wraz z nagraniem!
https://azurecommunity.pl/blog/azure-solutions-architect-governance
